### ISSUE TITLE
Is it possible to save the trained model as a Tensorflow frozen graph?

### ISSUE THREAD

**Issue by hodaraad** [jfkirk/tensorrec#91]
I saw in another issue the discussion around saving/loading the trained model and you added those abilities to the code. But, it seems they are based on Python pickle object. I'm wondering if you can give me some insight about how to save the trained graph as a TensorFlow graph as I need to use it in another non-Python environment (our production is in Ruby and the only way I can think of transferring the trained model from TensorRec to there is to use TensorFlow's save/load methods).

Thanks

**Comment by jfkirk** [comment]
Hey @hodaraad ! When you call "save_model()", one of the two output files is `tensorrec_session.cpkt`. This file contains the graph as well as the state of the variables. If you want to re-construct the model outside of Python, you could load in this TensorFlow session.

The `.pkl` file only contains the Python wrappers around the graph, so you won't need those once you're in Ruby.

I haven't worked with Ruby's TensorFlow API so, once you're there, I'm not sure what barriers you will run in to on that side.

If you give that a shot, let me know how it goes. I'll leave this issue open for a few days.

**Comment by hodaraad** [comment]
Thanks a lot for your prompt and helpful answer. That's great then. I'll try it with a simple model and will test it on our Ruby end and will let you know. Only once I figure out how to convert our data format to TensorRec' one.

**Comment by jfkirk** [comment]
@hodaraad That part may be tricky too -- there are quite a few inputs to the graph that will need to be in place. `input_utils.py` contains methods I use to manipulate the sparse input data into iterator objects, which may be helpful.

Good luck! Let me know if I can help.
